src/com/yetanalytics/lrs_admin_ui/handlers.cljs

```diff
@@ -540,3 +554,123 @@
+   {;; Set loading state
+    :db (reduce
+         (fn [db' vis-k]
+           (assoc-in db' [::db/status :loading vis-k] true))
+         db
+         include)
+    ;; make request
+    :fx [[:http-xhrio
+          {:method          :get
+           :uri             (httpfn/serv-uri
+                             server-host
+                             "/admin/status")
+           :params          (reduce-kv
+                             (fn [m k v]
+                               (assoc m (name k) v))
+                             {"include" include}
+                             params)
+           :format          (ajax/json-request-format)
+           :response-format (ajax/json-response-format {:keywords? true})
+           :on-success      [:status/set-data include]
+           :on-failure      [:status/server-error include]
+           :interceptors    [httpfn/add-jwt-interceptor]}]]}))
+
+(re-frame/reg-event-fx
+ :status/server-error
+ global-interceptors
+ (fn [{:keys [db]} [_ include error]]
+   {:db (reduce
+         (fn [db' k]
+           (update-in db' [::db/status :loading] dissoc k))
+         db
+         include)
+    :fx [[:dispatch
+          [:server-error error]]]}))
+
+(def status-dispatch-all
+  (into []
+        (for [status-query ["statement-count"
+                            "actor-count"
+                            "last-statement-stored"
+                            "timeline"
+                            "platform-frequency"]]
+          [:dispatch [:status/get-data [status-query]]])))
+
+(defmethod page-fx :status [_]
+  status-dispatch-all)
+
+(re-frame/reg-event-fx
+ :status/get-all-data
+ global-interceptors
+ (fn [_ _]
+   {:fx status-dispatch-all}))
+
+(re-frame/reg-event-fx
+ :status/set-data
+ global-interceptors
+ (fn [{:keys [db]} [_ include status-data]]
+   {:db (reduce
+         (fn [db' k]
+           (update-in db'
+                      [::db/status :loading]
+                      dissoc
+                      k))
+         (update-in db [::db/status :data] merge status-data)
+         include)}))
+
+(def timeline-control-fx
+  "Fx call that will refesh the timeline for a control change, debounced."
+  [:debounce-dispatch-later
+   {:key :status/timeline-control
+    :dispatch [:status/get-data ["timeline"]]
+    :ms 1000}])
+
+(re-frame/reg-event-fx
+ :status/set-timeline-unit
+ global-interceptors
+ (fn [{:keys [db]} [_ unit]]
+   {:db (assoc-in db [::db/status :params :timeline-unit] unit)
+    :fx [timeline-control-fx]}))
+
+(re-frame/reg-event-fx
+ :status/set-timeline-since
+ global-interceptors
+ (fn [{:keys [db]} [_ since-datetime-str]]
+   (try
+     {:db (assoc-in db
+                    [::db/status :params :timeline-since]
+                    (u/local-datetime->utc
+                     since-datetime-str))
+      :fx [timeline-control-fx]}
+     (catch js/Error _
+       (.log js/console
+             (str "Invalid timestamp " since-datetime-str " was ignored"))))))
```

Comment:
Just to make sure, if an exception was raised within the try-catch block, it's not going to trigger any of the effects, right?

Reply:
Correct. The return here is `nil`, in which case re-frame does nothing and makes no changes.